### PR TITLE
Optimization texture structure

### PR DESCRIPTION
**Pull request by GuoLei1990**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

- Delete `RenderColorTexture` and `RenderDepthTexture`, The type of`RenderTarget` property `colorTexture` and `depthTexture` adjust to Texture.

### CODE REVIEW

File: packages/rhi-webgl/src/GLTextureCubeMap.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { IPlatformTextureCubeMap, Logger, TextureCubeFace, TextureCubeMap, TextureFormat } from "@oasis-engine/core";
+import { IPlatformTextureCubeMap, Logger, TextureCubeFace, TextureCube, TextureFormat } from "@oasis-engine/core";
```
Comment (zhuxudong) [inline packages/rhi-webgl/src/GLTextureCubeMap.ts:1]: rename this file to `GLTextureCube` is better

File: packages/rhi-webgl/src/GLRenderTarget.ts [hunk 2]
```diff
@@ -207,6 +212,7 @@ export class GLRenderTarget implements IPlatformRenderTarget {
   }
 
   private _bindMainFBO(): void {
+    debugger;
```
Comment (zhuxudong) [inline packages/rhi-webgl/src/GLRenderTarget.ts:215]: ???

**Review by singlecoder — APPROVED** [review]
+1

**Review by zhuxudong — CHANGES_REQUESTED** [review]

**Review by zhuxudong — CHANGES_REQUESTED** [review]
`GLRenderColorTexture` , `RenderBufferDepthFormat`, `createPlatformTextureCubeMap` still have leftovers

File: packages/core/src/texture/RenderTarget.ts [hunk 3]
```diff
@@ -168,26 +179,24 @@ export class RenderTarget extends EngineObject {
...
       const colorTexture = this._colorTextures[i];
-      if (colorTexture.autoGenerateMipmaps) {
+      if (this.autoGenerateMipmaps) {
         colorTexture.generateMipmaps();
       }
     }
+
+    if (this.autoGenerateMipmaps) {
+      this._depthTexture && this._depthTexture.generateMipmaps();
+    }
```
Comment (zhuxudong) [inline packages/core/src/texture/RenderTarget.ts:199]: if(this.autoGenerateMipmaps){
... foreach colortexture
... depth texture

}

**Review by zhuxudong — CHANGES_REQUESTED** [review]

File: packages/loader/src/EnvLoader.ts [hunk 4]
```diff
@@ -6,10 +6,7 @@ import {
   Loader,
   LoadItem,
   resourceLoader,
-  ResourceManager,
-  TextureCubeFace,
-  TextureCubeMap,
-  TextureFilterMode
+  ResourceManager, TextureCube, TextureCubeFace, TextureFilterMode
```
Comment (zhuxudong) [inline packages/loader/src/EnvLoader.ts:9]: format code

**Review by zhuxudong — CHANGES_REQUESTED** [review]
`format` can move to `Texture` now

File: packages/loader/src/EnvLoader.ts (hunk 4 above)
Comment (GuoLei1990) [reply packages/loader/src/EnvLoader.ts]: done

**Comment by GuoLei1990** [comment]
> `format` can move to `Texture` now

done

File: packages/rhi-webgl/src/GLRenderTarget.ts [hunk 5]
```diff
@@ -45,24 +43,31 @@ export class GLRenderTarget implements IPlatformRenderTarget {
...
-    if (!(_depth instanceof RenderDepthTexture) && !GLTexture._supportRenderBufferDepthFormat(_depth, rhi, false)) {
-      throw new Error(`RenderBufferDepthFormat is not supported:${RenderBufferDepthFormat[_depth]}`);
+    for (let i = 0, n = _colorTextures.length; i < n; i++) {
+      const format = _colorTextures[i]._format;
+      if (!GLTexture._supportRenderBufferColorFormat(format, rhi)) {
+        throw new Error(`TextureFormat is not supported:${TextureFormat[format]} in RenderTarget`);
+      }
+    }
+
+    if (!GLTexture._supportRenderBufferDepthFormat(_depth, rhi, _depth instanceof Texture)) {
```
Comment (zhuxudong) [inline packages/rhi-webgl/src/GLRenderTarget.ts:53]: you must ensure `_depth` is `format` rather than `texture`

**Review by zhuxudong — CHANGES_REQUESTED** [review]

**Review by zhuxudong — APPROVED** [review]
+1

**Pull request merged**